core/asm/asm_test.go

```diff
@@ -55,6 +55,24 @@ func TestInstructionIteratorInvalid(t *testing.T) {
 	}
 }
 
+// Tests disassembling the instructions for PUSH0
+func TestInstructionIteratorPUSH0(t *testing.T) {
```

Comment:
I don't fully grokk how the test here is related to the bug or the fix, but that's fine I guess. However, I think these four tests might be sufficiently many to be replaced by one single table-driven one. I'll push a fix

Reply:
That was the other place where `IsPush` was used. I wanted to make sure I didn't break it :)